utils/src/main/java/com/redhat/lightblue/migrator/utils/DAOFacadeBase.java

```diff
@@ -0,0 +1,198 @@
+package com.redhat.lightblue.migrator.utils;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.redhat.lightblue.migrator.features.LightblueMigration;
+import com.redhat.lightblue.migrator.features.LightblueMigrationConfiguration;
+
+/**
+ * A helper base class for migrating services from legacy datastore to lightblue. It lets you call any service/dao method, using togglz switches to choose which service/dao to use and verifying
+ * returned data. Verification uses equals method.
+ *
+ * @author mpatercz
+ *
+ */
+@SuppressWarnings("all")
+public class DAOFacadeBase {
+
+    private static final Logger log = LoggerFactory.getLogger(LightblueMigrationConfiguration.class);
+
+    protected final Object legacyDAO, lightblueDAO;
+
+    public DAOFacadeBase(Object legacyDAO, Object lightblueDAO) {
```

Comment:
I'd make this an abstract class and use generics to get away from using Object.  I know it's just using reflection on these objects, but it will allow for some type checking, forcing legacyDAO and lightblueDAO to implement the same interface.